src/main/java/org/apache/pulsar/io/jcloud/partitioner/AbstractPartitioner.java

```diff
@@ -42,35 +46,67 @@ public abstract class AbstractPartitioner<T> implements Partitioner<T> {
     private boolean withTopicPartitionNumber;
     private boolean useIndexAsOffset;
 
+    private Map<String, String> topicsToPathMapping;
+    private Map<String, String> computedTopicsToPathMapping;
+
+
     @Override
     public void configure(BlobStoreAbstractConfig config) {
         this.sliceTopicPartitionPath = config.isSliceTopicPartitionPath();
         this.withTopicPartitionNumber = config.isWithTopicPartitionNumber();
         this.useIndexAsOffset = config.isPartitionerUseIndexAsOffset();
+        this.topicsToPathMapping = parseTopicsToPathMappingString(config.getTopicsToPathMapping());
+        this.computedTopicsToPathMapping = new HashMap<>();
+    }
+
+    private Map<String, String> parseTopicsToPathMappingString(String topicsToPathMappingString) {
+        Map<String, String> topicsToPathMapping = new HashMap<>();
+        if (StringUtils.isNotEmpty(topicsToPathMappingString)) {
+            for (String topicPathPair : topicsToPathMappingString.split(",")) {
+                String[] topicToPathMapping = topicPathPair.split("=");
+                if (topicToPathMapping.length == 2) {
+                    topicsToPathMapping.put(topicToPathMapping[0], topicToPathMapping[1]);
+                } else {
+                    LOGGER.warn("The topic to path mapping is not in correct format {}.", topicPathPair);
+                }
+            }
+        }
+        LOGGER.info("The final topics to path mapping is: {}", topicsToPathMapping);
+        return topicsToPathMapping;
     }
 
     @Override
     public String generatePartitionedPath(String topic, String encodedPartition) {
+        if (MapUtils.isNotEmpty(computedTopicsToPathMapping) && computedTopicsToPathMapping.containsKey(topic)) {
+            return StringUtils.join(Arrays.asList(computedTopicsToPathMapping.get(topic), encodedPartition),
```

Comment:
this is super costly for nothing,
can't be just 
String.format("%s%s%s", mapped, separator, encodedPartition)

?

Reply:
done